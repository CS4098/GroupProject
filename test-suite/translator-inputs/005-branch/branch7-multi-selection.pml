process test{
	branch br1 {
		selection sel1 {
	        action act1 {
		        requires { a }
		        provides { b }
	        }
	        action act2 {
		        requires { b }
		        provides { e }
	        }
	        action act3 {
		        requires { c }
		        provides { d }
	        }
    	}
   	    selection sel2 {
	        action act4 {
		        requires { a }
		        provides { d }
	        }
	        task t1 {
			    action act5 {
			        requires { a }
			        provides { b }
			    }
			    action act6 {
			        requires { b }
			        provides { c }
			    }
			    action act7 {
			        requires { c }
			        provides { d }
			    }
		    }
	    }
	    selection sel3 {
	        action act8 {
		        requires { a }
		        provides { x }
	        }
	        sequence seq1 {
			    action act9 {
			        requires { a }
	                provides { c }
			    }
	            action act10 {
			        requires { c }
	                provides { x }
			    }
		    }
	    }
	}
}
