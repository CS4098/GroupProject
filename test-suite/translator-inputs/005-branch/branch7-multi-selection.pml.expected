bool a;
bool b;
bool c;
bool d;
bool e;
bool x;

active proctype test()
{
	int br1 = _nr_pr;
	run sel1();
	run sel2();
    run sel3();
	_nr_pr == br1 ->
}

proctype sel1()
{
    if
    :: true ->
             a ->
                b = true;
    :: true ->
             b ->
                e = true;
    :: true ->
             c ->
                d = true;
    fi
}

proctype sel2()
{
    if
    :: true ->
            a ->
                d = true;
    :: true ->
                a ->
                    b = true;
                b ->
                    c = true;
                c ->
                    d = true;
    fi
}

proctype sel3()
{
    if
    :: true ->
            a ->
                x = true;
    :: true ->
                a ->
                    c = true;
                c ->
                    x = true;
    fi
}
